code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -570,3 +570,94 @@
+	ADD_TRAIT(affected_mob, TRAIT_NODEATH, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOHARDCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOSOFTCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_STABLEHEART, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOLIMBDISABLE, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_STUNRESISTANCE, FROM_NOOARTRIUM)
+	if(affected_mob.stat == DEAD)
+		back_from_the_dead = TRUE
+	affected_mob.set_stat(CONSCIOUS) //This doesn't touch knocked out
+	affected_mob.updatehealth()
+	affected_mob.update_sight()
+	affected_mob.add_movespeed_modifier(/datum/movespeed_modifier/reagent/nooartrium)
+	affected_mob.add_actionspeed_modifier(/datum/actionspeed_modifier/nooartrium)
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, STAT_TRAIT)
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, CRIT_HEALTH_TRAIT) //Because these are normally updated using set_health() - but we don't want to adjust health, and the addition of NOHARDCRIT blocks it being added after, but doesn't remove it if it was added before
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT) //Prevents the user from being knocked out by oxyloss
+	affected_mob.set_resting(FALSE) //Please get up, no one wants a deaththrows juggernaught that lies on the floor all the time
+	affected_mob.SetAllImmobility(0)
+	if(!back_from_the_dead)
+		to_chat(affected_mob, span_userdanger("You feel your heart start beating with incredible strength!"))
+		return
+	ADD_TRAIT(affected_mob, TRAIT_NOSTAMCRIT, FROM_NOOARTRIUM) // Moving corpses dont get tired
+	metabolization_rate = 0.4 * REM// Keeping dead corpse moving is harder
+	affected_mob.grab_ghost(force = FALSE) //Shoves them back into their freshly reanimated corpse.
+	affected_mob.emote("gasp")
+	to_chat(affected_mob, span_userdanger("You feel your heart start beating with incredible strength, forcing your battered body to move!"))
+
+
+/datum/reagent/drug/nooartrium/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
+	. = ..()
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, CRIT_HEALTH_TRAIT)
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)
+	affected_mob.adjustBruteLoss(0.5)
+	affected_mob.adjustOrganLoss(ORGAN_SLOT_HEART, 0.5)
+	if(back_from_the_dead)
+		affected_mob.adjustOrganLoss(ORGAN_SLOT_HEART, 0.25) // Keeping dead corpse moving is harder
+	var/obj/item/organ/heart/heart = affected_mob.getorganslot(ORGAN_SLOT_HEART)
+	if(!heart || heart.organ_flags & ORGAN_FAILING)
+		on_mob_delete(affected_mob)
+
+
+/datum/reagent/drug/nooartrium/on_mob_delete(mob/living/carbon/affected_mob)
+	. = ..()
+	remove_buffs(affected_mob)
+	var/obj/item/organ/heart/heart = affected_mob.getorganslot(ORGAN_SLOT_HEART)
+	if(affected_mob.health < -500 || heart.organ_flags & ORGAN_FAILING)//Honestly commendable if you get -500
+		explosion(affected_mob, light_impact_range = 1)
+		qdel(heart)
+		affected_mob.visible_message(span_boldwarning("[affected_mob]'s heart explodes!"))
```

Comment:
```suggestion
	if(back_from_the_dead || affected_mob.health < -300 || heart.organ_flags & ORGAN_FAILING)
		affected_mob.add_splatter_floor(get_turf(M))
		qdel(heart)
		affected_mob.visible_message(span_boldwarning("[affected_mob]'s heart explodes!"))
```

I don't think this should be a revival drug. If they were a corpse when it was used on them, they should go back to being a corpse but without a heart this time. 

Likewise, the heart exploding doesn't need to actually trigger an explosion,